docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,249 @@
+# Contributing
+
+Thanks for taking the time to join our community and start contributing.
+These guidelines will help you get started with the Contour Operator project.
+Please note that we require [DCO sign off](#dco-sign-off).  
+
+For more insight into the Contour Operator development workflow, reference the
+[how-we-work](https://projectcontour.io/resources/how-we-work/) page.
+
+## Building from source
+
+This section describes how to build Contour Operator from source.
+
+### Prerequisites
+
+1. *Install Go*
+
+    Contour Operator requires [Go 1.15][1] or later. We also assume that you're familiar with
+    Go's [`GOPATH` workspace][3] convention and have the appropriate environment variables set.
+
+### Fetch the source
+
+Contour Operator uses [`go modules`][2] for dependency management.
+
+```
+go get github.com/projectcontour/contour-operator
+```
+
+The remainder of this document assumes your terminal's working directory is
+`$GOPATH/src/github.com/projectcontour/contour-operator`.
+
+### Building
+
+To build Contour Operator, run:
+
+```
+make manager
+```
+
+This produces a `contour-operator` binary in your `$GOPATH/bin` directory and runs go fmt and go vet against the code.
+
+### Running the unit tests
+
+You can run all the unit tests for the project:
+
+```
+make test
+```
+
+To run the tests for a single package, change to package directory and run:
+
+```
+go test .
+```
+
+__Note:__ Unit tests must pass for your PR to get merged.
+
+## Contribution workflow
+
+This section describes the process for contributing a bug fix or new feature.
+
+### Before you submit a pull request
+
+This project operates according to the _talk, then code_ rule. If you plan to
+submit a pull request for anything more than a typo or obvious bug fix,
+first you _should_ [raise an issue][6] to discuss your proposal, before submitting any code.
```

Comment:
Seems like this PR technically violates this rule, but I won't hold you to it...